src/app/core/services/course-service/course.service.ts

```diff
@@ -1,53 +1,101 @@
+import { HttpClient, HttpParams } from "@angular/common/http";
 import { Injectable } from "@angular/core";
 import { BehaviorSubject, Observable, of } from "rxjs";
+import { filter, switchMap, tap, scan, startWith } from "rxjs/operators";
+import { API_URL } from "../../constants/index";
 
-import { COURSES } from "../../helpers/mock-courses";
-import { ICourse } from "../../models";
+
+import { ICourse } from "../../models/index";
 
 @Injectable()
 export class CourseService {
-    private courses: Array<ICourse> = COURSES;
+    private courses: Array<ICourse> = [];
+    private coursesPerPage = 5;
+    private http: HttpClient;
+    private coursesSubject: BehaviorSubject<Array<ICourse>> = new BehaviorSubject<Array<ICourse>>([]);
+
+    /**
+     *  subject that is used to implemnt pagination and load more functionality
+     */
+    public page$: BehaviorSubject<number>;
```

Comment:
better have public methods in service instead of calling next on subjects outside of the service directly